REFERENCE.md

```diff
@@ -0,0 +1,1099 @@
+# Reference
+<!-- DO NOT EDIT: This document was generated by Puppet Strings -->
+
+## Table of Contents
+
+**Classes**
+
+* [`python`](#python): Installs and manages python, python-dev, python-virtualenv and gunicorn.
+* [`python::config`](#pythonconfig): Optionally installs the gunicorn service
+* [`python::install`](#pythoninstall): Installs core python packages
+* [`python::params`](#pythonparams): The python Module default configuration settings.
+
+**Defined types**
+
+* [`python::dotfile`](#pythondotfile): Manages any python dotfiles with a simple config hash.
+* [`python::gunicorn`](#pythongunicorn): Manages Gunicorn virtual hosts.
+* [`python::pip`](#pythonpip): Installs and manages packages from pip.
+* [`python::pyvenv`](#pythonpyvenv): Create a Python3 virtualenv using pyvenv.
+* [`python::requirements`](#pythonrequirements): Installs and manages Python packages from requirements file.
+* [`python::virtualenv`](#pythonvirtualenv): Creates Python virtualenv.
+
+## Classes
+
+### python
+
+Installs and manages python, python-dev, python-virtualenv and gunicorn.
+
+#### Examples
+
+##### ensure system python is installed, with pip,dev, virtualenv, and gunicorn packages present
+
+```puppet
+class { 'python':
+  version    => 'system',
+  pip        => 'present',
+  dev        => 'present',
+  virtualenv => 'present',
+  gunicorn   => 'present',
+}
+```
+
+#### Parameters
+
+The following parameters are available in the `python` class.
+
+##### `ensure`
+
+Data type: `Enum['absent', 'present', 'latest']`
+
+Desired installation state for the Python package.
+Allowed values: absent, present and latest
+
+Default value: $python::params::ensure
+
+##### `version`
+
+Data type: `Any`
+
+Python version to install. Beware that valid values for this differ a) by the provider you choose and b) by the osfamily/operatingsystem you are using.
+Allowed values:
+ - provider == pip: everything pip allows as a version after the 'python=='
+ - else: 'system', 'pypy', 3/3.3/...
+    - Be aware that 'system' usually means python 2.X.
+    - 'pypy' actually lets us use pypy as python.
+    - 3/3.3/... means you are going to install the python3/python3.3/...
+      package, if available on your osfamily.
+
+Default value: $python::params::version
+
+##### `pip`
+
+Data type: `Enum['absent', 'present', 'latest']`
+
+Desired installation state for python-pip. Boolean values are deprecated.
+Allowed values: 'absent', 'present', 'latest'
+
+Default value: $python::params::pip
+
+##### `dev`
+
+Data type: `Any`
+
+Desired installation state for python-dev. Boolean values are deprecated.
+Allowed values: 'absent', 'present', 'latest'
+
+Default value: $python::params::dev
+
+##### `virtualenv`
+
+Data type: `Enum['absent', 'present', 'latest']`
+
+Desired installation state for python-virtualenv. Boolean values are deprecated
```

Comment:
With the data type this looks outdated. Could you remove the mention of boolean values and the allowed values because `Enum` already tells that.